starboard/core/votes.py

```diff
@@ -53,16 +53,14 @@ async def is_vote_valid_for(
     author_obj: hikari.Member | None,
     voter: hikari.Member,
 ) -> bool:
-    if (not config.self_vote) and voter.id == orig_message.author_id:
-        return False
-
-    if author.is_bot and not config.allow_bots:
-        return False
-
-    if orig_message.trashed:
-        return False
-
-    if orig_message.frozen:
+    if (
+        not config.self_vote
+        and voter.id == orig_message.author_id
+        or author.is_bot
+        and not config.allow_bots
+        or orig_message.trashed
+        or orig_message.frozen
+    ):
```

Comment:
For a few if-statements I think this is okay, but since there's so many it's a little bit painful to read.